### ISSUE TITLE
parseUrlHttp port not working

### ISSUE THREAD

**Issue by songpp** [mrkkrp/req#10]
```
main = do
  let (url, options) = fromJust (parseUrlHttp "http://127.0.0.1:7001/api")
  r <- req GET
        url
        NoReqBody
        jsonResponse
        options
  print (responseBody r :: Value)
```

output

```
*Req Data.ByteString> :main []
*** Exception: VanillaHttpException (HttpExceptionRequest Request {
  host                 = "127.0.0.1:7001"
  port                 = 80
  secure               = False
  requestHeaders       = []
  path                 = "/api"
  queryString          = ""
  method               = "GET"
  proxy                = Nothing
  rawBody              = False
  redirectCount        = 10
  responseTimeout      = ResponseTimeoutDefault
  requestVersion       = HTTP/1.1
}
 (ConnectionFailure getAddrInfo: does not exist (nodename nor servname provided, or not known)))
```

**Comment by mrkkrp** [comment]
Thanks for the report, I'll take a look soon.

**Comment by mrkkrp** [comment]
Indeed port parsing is not yet supported as `parseUrlHttp(s)` is primarily intended for use-cases when some API returns you URLs, and those typically do not have ports in them. It would be nice to be able to parse ports though.

In your case you can do just:

```haskell
main = do
  r <- req GET
        (http "localhost" /: "api")
        NoReqBody
        jsonResponse
        (port 7001)
  print (responseBody r :: Value)
```

But you probably know that already.

----

I'll keep this issue as a feature-request for port parsing in those helpers.

**Comment by mrkkrp** [comment]
The feature is now in `master`, will be available in Req 0.3.0.
